### PR TITLE
(feat) O3-3063: UI tweaks for the Today's appointments widget

### PR DESCRIPTION
**Pull request by denniskigen**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary
This PR applies various UI tweaks to the Today's Appointments widget to align it with other data tiles in O3.

### Before 

CleanShot 2024-04-09 at 9  42 56@2x

CleanShot 2024-04-09 at 9  36 02@2x

### After

CleanShot 2024-04-09 at 9  50 16@2x

CleanShot 2024-04-09 at 9  35 39@2x

## Related Issue

- [O3-3063](https://openmrs.atlassian.net/browse/O3-3063)
- [Slack conversation](https://openmrs.slack.com/archives/C039BLMGMMX/p1712688570518969)

### CODE REVIEW

**Comment by ojwanganto** [comment]
Thanks for working on this @denniskigen

**Review by ojwanganto — APPROVED** [review]

File: packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx [hunk 1]
```diff
@@ -77,23 +69,10 @@ const AppointmentActions: React.FC<AppointmentActionsProps> = ({ appointment })
   };
 
   return (
-    <div style={{ display: 'flex', alignItems: 'center' }}>
-      {renderVisitStatus()}
-      {isFutureAppointment || (isTodayAppointment && (!handleCheckout || !hasActiveVisitToday)) ? (
-        <OverflowMenu aria-label="Actions" iconDescription={t('actions', 'Actions')} size="sm" flipped>
```
Comment (denniskigen) [inline packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx:83]: - I don't know why we're checking for the absence of the `handleCheckout` function in this condition. It seems wrong to me. I've removed it from the conditional logic that determines whether to show the `OverflowMenu` or not.

**Comment by denniskigen** [comment]
State of things following my latest commit:

https://github.com/openmrs/openmrs-esm-patient-management/assets/8509731/aba23461-9b93-40d2-b703-173e01181f3b

**Comment by ojwanganto** [comment]
Looks good. Thanks @denniskigen

File: packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx (hunk 1 above)
Comment (chibongho) [reply packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx:83]: Wait... did you remove this OverflowMenu completely?

Comment (denniskigen) [reply packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx:83]: No, I moved it to `AppointmentsTable`

**Review by chibongho — APPROVED** [review]
Looks great. Thanks so much for doing this @denniskigen. Lots of subtle improvements. 

If we won't be standardizing tables like this into esm-styleguide, I wish there is at least a reference implementation somewhere (say esm-template-app) that we can just copy-paste from to get the styling right.

**Comment by denniskigen** [comment]
Sure, @chibongho. Definitely something I could work into `esm-template-app`. Thanks for the tip.

**Pull request merged**